lms/djangoapps/courseware/access.py

```diff
@@ -417,60 +417,72 @@ def _has_group_access(descriptor, user, course_key):
         log.warning("Group access check excludes all students, access will be denied.", exc_info=True)
         return ACCESS_DENIED
 
-    # resolve the partition IDs in group_access to actual
-    # partition objects, skipping those which contain empty group directives.
-    # If a referenced partition could not be found, it will be denied
-    # If the partition is found but is no longer active (meaning it's been disabled)
-    # then skip the access check for that partition.
-    partitions = []
-    for partition_id, group_ids in merged_access.items():
+    partition_groups = {}
+    user_groups = {}
+    if prefetched_group_data:
```

Comment:
all of this code looks right, you might consider wrapping this prefetch chunk into a helper function, in the interest of not making this function any longer than it already is.